wallet-new/src/Cardano/Wallet/API.hs

```diff
@@ -24,13 +27,37 @@ import qualified Cardano.Wallet.API.V1 as V1
 -- * 'Cardano.Wallet.Server' contains the main server;
 -- * 'Cardano.Wallet.API.V0.Handlers' contains all the @Handler@s serving the V0 API;
 -- * 'Cardano.Wallet.API.V1.Handlers' contains all the @Handler@s serving the V1 API;
+-- * 'Cardano.Wallet.API.Dev.Handlers' contains all the @Handler@s serving the Dev API;
 --
+-- type WalletAPI
+--     =     WalletPublicAPI
+--     :<|>  WalletDevAPI
+
+type V0_API
+     = "api" :> Tags '["V0 (Deprecated)"]
+             :> V0.API
+
+type V1_API
+     = "api" :> "v1"
+             :> Tags '["V1"]
+             :> V1.API
+
+type DEV_API
+     = "api" :> "development"
+             :> Tags '["Development"]
+             :> Dev.API
+
 type WalletAPI
-    =    "api" :> Tags '["V0 (Deprecated)"]
-               :> V0.API
-    :<|> "api" :> "v1"
-               :> Tags '["V1"]
-               :> V1.API
+    =     V0_API
+    :<|>  V1_API
+    :<|>  DEV_API
```

Comment:
This seems like a gratuitous refactoring -- what was wrong with the code before? It was nice and descriptive, now we added another layer of type alias a programmer needs to unwrap to understand what the API is doing (my 2 cents anyway).

Reply:
The first idea behind was to split API into different parts, so that we can hide endpoints in development mode. But I think your idea https://github.com/input-output-hk/cardano-sl/pull/2549/files#r169258990 is more elegant.